oauth/opts.go

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright (c) 2020 SSH Communications Security Inc.
+//
+// All rights reserved.
+//
+
+package oauth
+
+import (
+	"encoding/base64"
+	"io/ioutil"
+	"os"
+
+	"github.com/BurntSushi/toml"
+)
+
+// Option is configuration applied to the client
+type Option func(*tAuth) *tAuth
+
+// Access setups client access key
+func Access(access string) Option {
+	return func(auth *tAuth) *tAuth {
+		if access != "" {
+			auth.access = access
+		}
+		return auth
+	}
+}
+
+// Secret setups clients secret key
+func Secret(secret string) Option {
+	return func(auth *tAuth) *tAuth {
+		if secret != "" {
+			auth.secret = secret
+		}
+		return auth
+	}
+}
+
+// Digest setups client secret digest
+func Digest(oauthAccess, oauthSecret string) Option {
+	return func(auth *tAuth) *tAuth {
+		if oauthAccess != "" && oauthSecret != "" {
+			auth.digest = base64.StdEncoding.EncodeToString([]byte(oauthAccess + ":" + oauthSecret))
+		}
+		return auth
+	}
+}
+
+// UseConfigFile setup credential from tol file
+func UseConfigFile(path *string) Option {
```

Comment:
Is there a reason for passing the path as reference? This is not in-line with other Option functions.

Reply:
Nop, left over before cefabba commit